tests/integration_tests/modules/test_apt.py

```diff
@@ -200,30 +200,41 @@ def test_os_pipelining(self, class_client: IntegrationInstance):
         assert conf_exists is False
 
 
-DEFAULT_DATA = """\
+_DEFAULT_DATA = """\
 #cloud-config
 apt:
   primary:
     - arches:
       - default
+      {uri}
   security:
     - arches:
       - default
 """
+DEFAULT_DATA = _DEFAULT_DATA.format(uri='')
 
 
 @pytest.mark.ubuntu
 @pytest.mark.user_data(DEFAULT_DATA)
 class TestDefaults:
-    def test_primary(self, class_client: IntegrationInstance):
-        """Test apt default primary sources.
+    @pytest.mark.openstack
+    def test_primary_on_openstack(self, class_client: IntegrationInstance):
+        """Test apt default primary source on openstack.
 
-        Ported from
-        tests/cloud_tests/testcases/modules/apt_configure_primary.py
+        When no uri is provided.
         """
+        version = class_client.execute(
+            'curl http://169.254.169.254'
+        ).split('\n')[-1]
+        zone = class_client.execute(
+            'curl http://169.254.169.254/{}/meta-data/placement/'
+            'availability-zone'.format(version),
+        )
```

Comment:
Could we use `cloud-init query v1.availability_zone` here instead?

Reply:
Generalising it is out-of-scope for this PR, but if we do, can this test run on more than just OpenStack? We perform region/AZ substitution on every cloud; an EC2 instance I have kicking around, for example:

```
deb http://eu-west-1.ec2.archive.ubuntu.com/ubuntu/ focal main restricted
```

Reply:
Yep, I noticed there there were different templates for different clouds, so I figured we would want multiple versions, but since I was just testing on openstack, I started with openstack.